microovn/ovn/cmd/commands.go

```diff
@@ -38,23 +40,69 @@ type OvsdbSpec struct {
 type OvsdbType int
 
 const (
-	// OvsdbTypeNBLocal     - OVSDB Database with schema OVN_Northbound.
+	// OvsdbTypeNBLocal     - OVSDB Database with local unix connection and OVN_Northbound schema.
 	OvsdbTypeNBLocal OvsdbType = iota
-	// OvsdbTypeSBLocal     - OVSDB Database with schema OVN_Southbound.
+	// OvsdbTypeNBCluster   - OVSDB Database with network connection (tcp/ssl) and OVN_Northbound schema.
+	OvsdbTypeNBCluster
+	// OvsdbTypeSBLocal     - OVSDB Database with local unix connection and OVN_Southbound schema.
 	OvsdbTypeSBLocal
 	// OvsdbTypeSwitchLocal - OVSDB Database with schema Open_vSwitch.
 	OvsdbTypeSwitchLocal
 )
 
+// loadConnectionString returns string that defines network connection to OVSDB. Information
+// is sourced from the OVN environment file (paths.OvnEnvFile).
+// Example of returned value: "ssl:10.0.0.1:6641,ssl:10.0.0.2:6641,ssl:10.0.0.3:6641"
+func loadConnectionString(dbType OvsdbType) (string, error) {
+	var envVar string
+	if dbType == OvsdbTypeNBCluster {
+		envVar = "OVN_NB_CONNECT"
+	} else {
+		return "", errors.New("failed to load connection string: unknown ovsdb type")
+	}
+
+	file, err := os.Open(paths.OvnEnvFile())
+	if err != nil {
+		return "", fmt.Errorf("error opening OVN env file: %s", err)
+	}
+	defer file.Close()
+
+	var connectionString string
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		key, value, found := strings.Cut(line, "=")
+		if found && key == envVar {
+			connectionString = value
+			break
+		}
+	}
+
+	if connectionString == "" {
+		return "", fmt.Errorf("connection string for %s not found in OVN env file", envVar)
+	}
+	return strings.Trim(connectionString, "\""), nil
+}
+
 // NewOvsdbSpec is a helper function that takes OvsdbType as an argument and generates
 // proper OvsdbSpec for given type.
 func NewOvsdbSpec(dbType OvsdbType) (*OvsdbSpec, error) {
 	var dbSpec *OvsdbSpec
 	var err error
 
-	if dbType == OvsdbTypeNBLocal {
+	if dbType == OvsdbTypeNBLocal || dbType == OvsdbTypeNBCluster {
+		var url string
+		if dbType == OvsdbTypeNBLocal {
+			url = fmt.Sprintf("unix:%s", paths.OvnNBDatabaseSock())
```

Comment:
Yes, the Local/Cluster versions of Northbound differ only in the socket, so I didn't want to copy-paste the entire struct initialization. Golang does not support ternary operators, so the only way that I can think of to "shorten" this would be
```golang
url := fmt.Sprintf("unix:%s", paths.OvnNBDatabaseSock())
if dbType == OvsdbTypeNBCluster {
    	url, err = loadConnectionString(dbType)
    	if err != nil {
    	    return nil, err
    	}
}
```
However with your suggestion above to use command wrappers to talk to clustered OVN NB, this may get removed entirely.

Reply:
Removed entirely in favor of using `ovn-nbctl` wrapper.